File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        int n = happiness.length;
        Arrays.sort(happiness);
        long temp = 0;
        
        for(int i=0; i<k; i++)
        {
            if(happiness[n-i-1] - i < 0) break;
            temp = temp + (happiness[n-i-1]) - i;
        }
        return temp;
    }
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        n = len(happiness)
        happiness.sort()
        temp = 0
        
        for i in range(k): 
            if happiness[n-i-1] - i < 0: break
            temp = temp + (happiness[n-i-1]) - i
        
        return temp